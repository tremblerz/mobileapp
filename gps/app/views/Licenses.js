import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import {
  BackHandler,
  Image,
  Linking,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';
import { WebView } from 'react-native-webview';

import foreArrow from '../../../shared/assets/images/foreArrow.png';
import licenses from '../../../shared/assets/LICENSE.json';
import NavigationBarWrapper from '../components/NavigationBarWrapper';
import { Typography } from '../components/Typography';
import Colors from '../constants/colors';
import { Theme } from '../constants/themes';

const PRIVACY_POLICY_URL =
  'https://docs.google.com/document/d/17u0f8ni9S0D4w8RCUlMMqxAlXKJAd2oiYGP8NUwkINo/edit';

export const LicensesScreen = ({ navigation }) => {
  const { t } = useTranslation();

  const backToMain = () => {
    navigation.goBack();
  };

  const handleBackPress = () => {
    backToMain();
    return true;
  };

  const handleTermsOfUsePressed = () => {
    Linking.openURL(PRIVACY_POLICY_URL);
  };

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', handleBackPress);
    return () => {
      BackHandler.removeEventListener('hardwareBackPress', handleBackPress);
    };
  });

  function getLicenses() {
    let result = '<html>';
    result +=
      '<style>  html, body { font-size: 40px; margin: 0; padding: 0; } </style>';
    result += '<body>';

    for (let i = 0; i < licenses.terms_and_licenses.length; i++) {
      const element = licenses.terms_and_licenses[i];

      result += '<H2>' + element.name + '</H2><P>';
      result += element.text.replace(/\n/g, '<br/>');
      result += '<hr/>';
    }
    result += '</body></html>';

    return result;
  }

  return (
    <NavigationBarWrapper
      title={t('label.legal_page_title')}
      onBackPress={backToMain}>
      <ScrollView contentContainerStyle={styles.contentContainer}>
        <View style={{ flex: 4 }}>
          <WebView
            originWhitelist={['*']}
            source={{
              html: getLicenses(),
            }}
            style={{
              marginTop: 15,
              backgroundColor: Colors.INTRO_WHITE_BG,
            }}
          />
        </View>
      </ScrollView>
      <Theme use='charcoal'>
        <TouchableOpacity
          onPress={handleTermsOfUsePressed}
          style={styles.termsInfoRow}>
          <Typography
            use='headline2'
            onPress={() => Linking.openURL(PRIVACY_POLICY_URL)}>
            {t('label.privacy_policy')}
          </Typography>
          <View style={styles.arrowContainer}>
            <Image source={foreArrow} />
          </View>
        </TouchableOpacity>
      </Theme>
    </NavigationBarWrapper>
  );
};

const styles = StyleSheet.create({
  contentContainer: {
    flexDirection: 'column',
    width: '100%',
    backgroundColor: Colors.INTRO_WHITE_BG,
    paddingHorizontal: 26,
    flex: 1,
  },
  termsInfoRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: Colors.SILVER,
    padding: 15,
  },
  arrowContainer: {
    alignSelf: 'center',
    paddingRight: 20,
    paddingLeft: 20,
  },
});
