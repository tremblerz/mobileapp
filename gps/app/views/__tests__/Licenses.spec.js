import 'react-native';

import { render } from '@testing-library/react-native';
import React from 'react';

import { LicensesScreen } from '../Licenses';

it('renders correctly', () => {
  const { asJSON } = render(<LicensesScreen />);

  expect(asJSON()).toMatchSnapshot();
});
