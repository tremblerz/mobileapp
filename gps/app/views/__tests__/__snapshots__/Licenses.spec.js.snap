// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#4051DB",
        "flexBasis": 0,
        "flexGrow": 0,
        "flexShrink": 1,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#F7F8FF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4051DB",
          "borderBottomColor": "#5061E6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "alignItems": "center",
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "width": 60,
            "zIndex": 1,
          }
        }
      >
        <SvgXml
          style={
            Object {
              "color": "#FFF",
              "height": 18,
              "opacity": 0.4,
              "width": 18,
            }
          }
          xml="
  <svg width=\\"23\\" height=\\"24\\" viewBox=\\"0 0 23 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M11.4854 3L2.22803 12.2573L11.4854 21.5146\\" stroke=\\"currentColor\\" stroke-opacity=\\"0.4\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" />
  </svg>
"
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Medium",
            "fontSize": 26,
            "letterSpacing": -0.3,
            "lineHeight": 34,
            "paddingHorizontal": 20,
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        Legal
      </Text>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "backgroundColor": "#F7F8FF",
          "flex": 1,
          "flexDirection": "column",
          "paddingHorizontal": 26,
          "width": "100%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 4,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <RNCWebView
              cacheEnabled={true}
              javaScriptEnabled={true}
              messagingEnabled={false}
              source={
                Object {
                  "html": "<html><style>  html, body { font-size: 40px; margin: 0; padding: 0; } </style><body><H2>COVID Safe Paths</H2><P>MIT License<br/><br/>Copyright (c) 2020 TripleBlind, Inc<br/><br/>Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the \\"Software\\"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:<br/><br/>The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.<br/><br/>THE SOFTWARE IS PROVIDED \\"AS IS\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.<br/><br/>Path Check, Inc.<br/>58 Day Street<br/>Box 441621<br/>Somerville, MA 02144<br/>info@pathcheck.org<br/>covidsafepaths.org<hr/></body></html>",
                }
              }
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "backgroundColor": "#F7F8FF",
                    "marginTop": 15,
                  },
                ]
              }
              useSharedProcessPool={true}
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      style={
        Object {
          "backgroundColor": "#BEBEBE",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Bold",
            "fontSize": 26,
            "fontWeight": "bold",
            "lineHeight": 34,
            "writingDirection": "ltr",
          }
        }
        use="headline2"
      >
        Privacy Policy
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../shared/assets/images/foreArrow.png",
            }
          }
        />
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
